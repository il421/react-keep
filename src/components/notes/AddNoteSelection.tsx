import React from "react";

export class AddNoteSelection extends React.PureComponent {
  render() {
    return (
      <>
        AddNoteSelection
      </>
    );
  }
}
