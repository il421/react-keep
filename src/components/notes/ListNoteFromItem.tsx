import React from "react";
import { FlexBox } from "../ui-components/FlexBox";
import { AlignItems, JustifyContent } from "../../common/variables";
import { CheckboxInputField } from "../form/CheckboxInputField";
import { TextInputField } from "../form/TextInputField";
import { IconButton } from "../ui-components/IconButton";
import "../../styles/components/notes/_list-note-form-item.scss";

interface ListNoteFromItemProps {
  name: string;
  onRemove: () => void;
}

export const ListNoteFromItem: React.FunctionComponent<ListNoteFromItemProps> = ({
  name,
  onRemove
}) => {
  return (
    <FlexBox
      justifyContent={JustifyContent.spaceBetween}
      alignItems={AlignItems.start}
      className="list-note-form-item"
    >
      {/* @TODO for drag and drop */}
      {/*<FontAwesomeIcon*/}
      {/*  icon={"th"}*/}
      {/*  className="list-note-form-item__mover"*/}
      {/*  size="lg"*/}
      {/*/>*/}
      <CheckboxInputField
        name={`${name}.checked`}
        className="list-note-form-item__check"
        id={`${name}.checked`}
      />
      <TextInputField
        isTextArea={true}
        name={`${name}.content`}
        placeholder="Type something ..."
        className="list-note-form-item__text"
      />
      <IconButton
        onButtonClick={onRemove}
        icon={"times"}
        className="list-note-form-item__remove"
      />
    </FlexBox>
  );
};
