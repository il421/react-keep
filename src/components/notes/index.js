export * from "./NotesList";
export * from "./NotesItem";
export * from "./notes.types";
export * from "./TextNoteFormModal";
export * from "./ListNoteFormModal";
export * from "./NoteForm";
export * from "./options/ColorsPickerField";
