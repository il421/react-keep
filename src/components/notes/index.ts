export * from "./ConfirmDialog";
export * from "./ImageNoteFormModal";
export * from "./ListNoteFormModal";
export * from "./ListNoteFromItem";
export * from "./NoteContent";
export * from "./NoteForm";
export * from "./notes.types";
export * from "./NotesItem";
export * from "./TextNoteFormModal";
export * from "./utils";
export * from "./options/ColorsPickerField";
export * from "./options/TagsPickerField";
