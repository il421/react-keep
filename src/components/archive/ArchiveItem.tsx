import React from "react";
import { ImageItem, ListItem, Note } from "../../store/store.types";
import { FlexBox, IconButton } from "../ui-components";
import "../../styles/components/archive/_archive-item.scss";
import { AlignItems, JustifyContent, getShortText } from "../../common";
import moment from "moment";
import { NoteType } from "../notes/notes.types";

export interface ArchiveItemProps {
  note: Note;
  unarchiveNote: (id: string) => void;
}

export const ArchiveItem: React.FunctionComponent<ArchiveItemProps> = ({
  note,
  unarchiveNote,
}) => {
  const NO_IMAGE_URL: string = "img/no_image.png";
  return (
    <FlexBox
      justifyContent={JustifyContent.spaceBetween}
      alignItems={AlignItems.baseline}
      className="archive-item"
    >
      <FlexBox
        justifyContent={JustifyContent.spaceBetween}
        className="archive-item__content"
      >
        <FlexBox
          className="archive-item__wrapper"
          vertical
          justifyContent={JustifyContent.spaceBetween}
        >
          <FlexBox
            className="archive-item-content"
            justifyContent={JustifyContent.spaceBetween}
          >
            <b className="archive-item-content__title">Title:</b>
            <div className="archive-item-content__value">{note.title}</div>
          </FlexBox>

          <FlexBox
            className="archive-item-content"
            justifyContent={JustifyContent.spaceBetween}
          >
            <b className="archive-item-content__title">Text:</b>
            <div className="archive-item-content__value">
              {note.type === NoteType.text ? (
                <div>{getShortText(note.content as string, 20)}</div>
              ) : note.type === NoteType.list ? (
                <ul className="content__list list">
                  {/* show 3 lines */}
                  {(note.content as ListItem[])
                    .slice(0, 3)
                    .map((item, index) => (
                      <li key={index}>
                        <div
                          className={`list__box ${
                            item.checked && "list__box--checked"
                          }`}
                        />
                        <span className="list__text">
                          {getShortText(item.content, 10)}
                        </span>
                      </li>
                    ))}
                </ul>
              ) : (
                <div>
                  <div>
                    {getShortText((note.content as ImageItem).content, 20)}
                  </div>
                  <img
                    src={
                      (note.content as ImageItem).imageUrl !== null
                        ? (note.content as ImageItem).imageUrl!
                        : NO_IMAGE_URL
                    }
                    alt="Picture of the note"
                  />
                </div>
              )}
            </div>
          </FlexBox>

          <FlexBox
            className="archive-item-content"
            justifyContent={JustifyContent.spaceBetween}
          >
            <b className="archive-item-content__title">Created:</b>
            <div className="archive-item-content__value">
              {moment(note.createdAt).format("MMMM Do, YYYY")}
            </div>
          </FlexBox>
        </FlexBox>
      </FlexBox>

      <IconButton
        className="archive-item__icon"
        icon="arrow-right"
        onButtonClick={() => unarchiveNote(note.id)}
      />
    </FlexBox>
  );
};
