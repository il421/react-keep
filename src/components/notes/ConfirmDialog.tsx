import React, { Dispatch } from "react";
import { connect } from "react-redux";
import { handleRemoveNote } from "../../actions/notes";

/**
 * This component works only as a part of Note for now.
 */

interface ConfirmDialogProps extends Pick<HTMLElement, "className">{
  id: string;
  removeNote: (id: string) => void;
  closeDialog: () => void;
}

const ConfirmDialog: React.FunctionComponent<ConfirmDialogProps> = ({ className, closeDialog, removeNote, id }) => {
  return (
    <div className={className}>
      <button type="button"
        onClick={() => {
          removeNote(id);
          closeDialog();
        }
        }
      >Yes</button>
      <button type="button" onClick={closeDialog}>No</button>
    </div>
  );
};


const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  removeNote: (id: string) => dispatch(handleRemoveNote(id))
});


export default connect(undefined, mapDispatchToProps)(ConfirmDialog);
