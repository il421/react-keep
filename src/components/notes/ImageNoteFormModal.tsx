import React from "react";
import Modal from "react-modal";
import { NoteFormValues, NoteType } from "./notes.types";
import {
  isModal,
  nameOf,
  Placeholders,
  AlignItems,
  JustifyContent,
  Errors,
} from "../../common";
import { PathNames, QueryKeys } from "../../routers/Routing";
import { FileFormField, TextInputField } from "../form";
import "../../styles/components/notes/_note-form.scss";
import "../../styles/components/notes/_image-note-form-item.scss";
import { NoteForm } from "./NoteForm";
import { History } from "history";
import { ImageItem } from "../../store/store.types";
import { Field, FormRenderProps, FormSpy } from "react-final-form";
import { IconButton, FlexBox } from "../ui-components";
import { ValidationErrors } from "final-form";

export interface ImageNoteFormModalProps {
  history: History;
}

export const getValidationErrors = (
  values: Pick<NoteFormValues<ImageItem>, "content">
): ValidationErrors => {
  const errors: ValidationErrors = { content: {} };
  const { uploadedImage, imageUrl } = values.content!;

  if (!uploadedImage && !imageUrl) {
    errors.content.uploadedImage = Errors.required;
  }

  return errors;
};

export class ImageNoteFormModal extends React.Component<
  ImageNoteFormModalProps
> {
  private nameOf = nameOf<NoteFormValues<ImageItem>>();
  private nameOfContent = nameOf<ImageItem>();
  private readonly NO_IMAGE_URL: string = "img/no_image.png";
  private readonly FIELD_ID: string = "image";

  private handleUploadClick = () => {
    const fileInput = document.getElementById(this.FIELD_ID);
    fileInput && fileInput.click();
  };

  render() {
    if (
      !isModal({
        query: this.props.history.location.search,
        type: QueryKeys.image,
      })
    ) {
      return null;
    }

    return (
      <Modal
        isOpen={true}
        onRequestClose={() => this.props.history.push(PathNames.base)}
        closeTimeoutMS={200}
        className="note-modal"
        ariaHideApp={false}
      >
        <NoteForm
          type={NoteType.image}
          history={this.props.history}
          validate={getValidationErrors}
        >
          <FormSpy>
            {({
              form,
              values,
              initialValues,
            }: FormRenderProps<NoteFormValues<ImageItem>>) => {
              return (
                <div className="image-note-form-item">
                  <Field
                    name={`${this.nameOf("content")}.${this.nameOfContent(
                      "imageUrl"
                    )}`}
                  >
                    {() => (
                      <img
                        className="image-note-form-item__img"
                        src={
                          values.content.imageUrl &&
                          !values.content.uploadedImage
                            ? values.content.imageUrl
                            : values.content.uploadedImage
                            ? URL.createObjectURL(values.content.uploadedImage)
                            : this.NO_IMAGE_URL
                        }
                        alt="Uploaded file"
                      />
                    )}
                  </Field>
                  {!initialValues.createdBy && (
                    <FlexBox
                      className="image-note-form-item__btns"
                      justifyContent={JustifyContent.spaceBetween}
                      alignItems={AlignItems.center}
                      vertical
                    >
                      <IconButton
                        icon="upload"
                        onButtonClick={() => {
                          this.handleUploadClick();
                        }}
                      />
                      <IconButton
                        icon="times"
                        onButtonClick={() => {
                          form.change(
                            `${this.nameOf("content")}[${this.nameOfContent(
                              "uploadedImage"
                            )}]`,
                            undefined
                          );
                        }}
                      />
                      {initialValues.content.imageUrl && (
                        <IconButton
                          id="test-trash-image-button"
                          icon="trash"
                          onButtonClick={() => {
                            form.change(
                              `${this.nameOf("content")}[${this.nameOfContent(
                                "imageUrl"
                              )}]`,
                              null
                            );
                            form.change(
                              `${this.nameOf("content")}[${this.nameOfContent(
                                "uploadedImage"
                              )}]`,
                              undefined
                            );
                          }}
                        />
                      )}
                      <FileFormField
                        id={this.FIELD_ID}
                        name={`${this.nameOf("content")}.${this.nameOfContent(
                          "uploadedImage"
                        )}]`}
                        className="image-note-form-item__field"
                      />
                    </FlexBox>
                  )}
                </div>
              );
            }}
          </FormSpy>

          <TextInputField
            isTextArea={true}
            name={`${this.nameOf("content")}.${this.nameOfContent("text")}`}
            placeholder={Placeholders.content}
            className="note-form__text note-form__text--image"
          />
        </NoteForm>
      </Modal>
    );
  }
}
