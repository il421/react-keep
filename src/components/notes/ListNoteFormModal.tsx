import React from "react";
import Modal from "react-modal";
import { ListItem } from "../../store/store.types";
import { NoteFormValues, NoteType } from "./notes.types";
import { PathNames, QueryKeys, RouteActions } from "../../routers/Routing";
import { NoteForm } from "./NoteForm";
import { History } from "history";
import { FieldArray, FieldArrayRenderProps } from "react-final-form-arrays";

import { ListNoteFromItem } from "./ListNoteFromItem";
import { FormRenderProps, FormSpy } from "react-final-form";
import { FlexBox, IconButton } from "../ui-components";
import {
  AlignItems,
  isModal,
  JustifyContent,
  nameOf,
  newLineRegEx,
} from "../../common";
import { getDefaultContent } from "./utils";
import { parse } from "query-string";

export interface ListNoteFormModalProps {
  history: History;
}

export class ListNoteFormModal extends React.Component<ListNoteFormModalProps> {
  private nameOf = nameOf<NoteFormValues<ListItem[]>>();

  private getCheckedItems = (content: ListItem[]): string => {
    if (!content) return "";
    return `${content.length}/${content.filter((c) => c.checked).length}`;
  };

  private isNewNote: boolean =
    parse(this.props.history.location.search)[QueryKeys.list] ===
    RouteActions.add;

  render() {
    if (
      !isModal({
        query: this.props.history.location.search,
        type: QueryKeys.list,
      })
    ) {
      return null;
    }
    return (
      <Modal
        isOpen={true}
        onRequestClose={() => this.props.history.push(PathNames.base)}
        closeTimeoutMS={200}
        className="note-modal"
        ariaHideApp={false}
      >
        <NoteForm type={NoteType.list} history={this.props.history}>
          <FormSpy>
            {({
              form,
              values,
            }: FormRenderProps<NoteFormValues<ListItem[]>>) => {
              const addItem = () => {
                form.change(this.nameOf("content"), [
                  getDefaultContent<ListItem[]>(NoteType.list)[0],
                  ...values.content,
                ]);
              };
              return (
                <>
                  <FlexBox
                    justifyContent={JustifyContent.spaceBetween}
                    alignItems={AlignItems.center}
                  >
                    <div id="test-checked-items">
                      Checked: {this.getCheckedItems(values.content)}
                    </div>
                    <IconButton
                      onButtonClick={addItem}
                      icon="plus-circle"
                      text="Add line"
                      id="test-add-item-button"
                    />
                  </FlexBox>

                  <div className="note-form__list">
                    <FieldArray name={this.nameOf("content")}>
                      {(
                        props: FieldArrayRenderProps<ListItem, HTMLElement> & {
                          fields: {
                            sortArray: (
                              compareFn: (a: any, b: any) => number
                            ) => void;
                          };
                        }
                      ) => {
                        const handleOnChecked = () => {
                          props.fields.sortArray(
                            (a: ListItem, b: ListItem) =>
                              Number(a.checked) - Number(b.checked)
                          );
                        };

                        const handleOnRemove = (index: number) => () => {
                          props.fields.remove(index);
                        };

                        const setPastedValue = (index: number) => (
                          paste: string
                        ) => {
                          const values: string[] = paste
                            .split(newLineRegEx)
                            .filter((v) => !!v);

                          if (values.length > 1) {
                            const defaultItem: ListItem = getDefaultContent<
                              ListItem[]
                            >(NoteType.list)[0];
                            form.batch(() => {
                              for (let i = 0; i < values.length; i++) {
                                if (i === 0) {
                                  // update the first element
                                  props.fields.update(index, {
                                    ...defaultItem,
                                    content: values[i].trim(),
                                  });
                                } else {
                                  // push others
                                  props.fields.push({
                                    ...defaultItem,
                                    content: values[i].trim(),
                                  });
                                }
                              }
                            });
                          }
                        };

                        return props.fields.map(
                          (name: string, index: number) => (
                            <ListNoteFromItem
                              autoFocus={this.isNewNote && index === 0}
                              name={name}
                              key={props.fields.length! - index} // to focus the first element
                              index={index}
                              onRemove={handleOnRemove(index)}
                              addItem={addItem}
                              onChecked={handleOnChecked}
                              setPastedValue={setPastedValue(index)}
                            />
                          )
                        );
                      }}
                    </FieldArray>
                  </div>
                </>
              );
            }}
          </FormSpy>
        </NoteForm>
      </Modal>
    );
  }
}
