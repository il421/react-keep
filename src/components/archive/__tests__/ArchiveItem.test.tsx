import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { ArchiveItem, ArchiveItemProps } from "../ArchiveItem";
import { notes } from "../../../testData/notes";

let props: ArchiveItemProps, wrapper: ShallowWrapper<ArchiveItemProps, any>;

beforeEach(() => {
  props = {
    note: notes[0],
    unarchiveNote: jest.fn(),
  };

  wrapper = shallow<ArchiveItemProps>(<ArchiveItem {...props} />);
});

test("should render ArchiveItem", () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

test("should unarchive note if click on button", () => {
  wrapper.find("IconButton").at(0).prop<any>("onButtonClick")();
  expect(props.unarchiveNote).toHaveBeenLastCalledWith(notes[0].id);
});
