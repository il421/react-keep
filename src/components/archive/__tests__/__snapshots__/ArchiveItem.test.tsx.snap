// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ArchiveItem 1`] = `
"<FlexBox justifyContent=\\"space-between\\" alignItems=\\"baseline\\" className=\\"archive-item\\">
  <FlexBox justifyContent=\\"space-between\\" flexGrow={true} className=\\"archive-item__content\\">
    <FlexBox className=\\"archive-item__wrapper\\" vertical={true} justifyContent=\\"space-between\\">
      <FlexBox className=\\"archive-item-content\\" justifyContent=\\"space-between\\">
        <b className=\\"archive-item-content__title\\">
          Title:
        </b>
        <div className=\\"archive-item-content__value\\">
          1
        </div>
      </FlexBox>
      <FlexBox className=\\"archive-item-content\\" justifyContent=\\"space-between\\">
        <b className=\\"archive-item-content__title\\">
          Text:
        </b>
        <div className=\\"archive-item-content__value\\">
          <div>
            text
          </div>
        </div>
      </FlexBox>
      <FlexBox className=\\"archive-item-content\\" justifyContent=\\"space-between\\">
        <b className=\\"archive-item-content__title\\">
          Created:
        </b>
        <div className=\\"archive-item-content__value\\">
          January 1st, 1970
        </div>
      </FlexBox>
    </FlexBox>
  </FlexBox>
  <IconButton className=\\"archive-item__icon\\" icon=\\"arrow-right\\" onButtonClick={[Function: onButtonClick]} />
</FlexBox>"
`;
